packages/p2p/src/stores/general-store.js

```diff
@@ -201,15 +201,20 @@ export default class GeneralStore extends BaseStore {
         this.updateP2pNotifications(notifications);
     }
 
+    @action.bound
     showCompletedOrderNotification(advertiser_name, order_id) {
+        const { order_store } = this.root_store;
         const notification_key = `order-${order_id}`;
 
         this.props.addNotificationMessage({
             action: {
                 onClick: () => {
+                    if (order_store.order_id === order_id) {
+                        order_store.setIsRatingModalOpen(true);
+                    }
```

Comment:
have to check if order_id has been set already or else if the user is not on order_details page and click the Give Feedback button, it'll redirect to order_details and the rating modal will open automatically